app/code/Magento/InventoryCatalog/Test/Integration/InventoryCatalog/Model/Source/Validator/PreventDisableValidatorTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Test\Integration\InventoryCatalog\Model\Source\Validator;
+
+use Magento\Framework\Webapi\Exception;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\TestFramework\TestCase\WebapiAbstract;
+
+class PreventDisableValidatorTest extends WebapiAbstract
+{
+    /**#@+
+     * Service constants
+     */
+    const RESOURCE_PATH = '/V1/inventory/source';
+    const SERVICE_NAME = 'inventoryApiSourceRepositoryV1';
+    /**#@-*/
+
+    /**
+     * @var array
+     */
+    private $validData = [
+        SourceInterface::SOURCE_CODE => 'source-code-1',
+        SourceInterface::NAME => 'source-name-1',
+        SourceInterface::POSTCODE => 'source-postcode',
+        SourceInterface::COUNTRY_ID => 'US',
+    ];
+
+    /**
+     * @param string $field
+     * @param $value
+     * @param array $expectedErrorData
+     * @dataProvider failedValidationDataProvider
+     * @throws \Exception
+     */
+    public function testFailedValidationOnUpdate(string $field, $value, array $expectedErrorData)
+    {
+        $data = $this->validData;
+        $data[$field] = $value;
+
+        $sourceCode = 'default';
```

Comment:
Pls, use corresponding service for get default source code

Reply:
I'm not sure I get your point. Could you explain?

Reply:
Does it mean that I should Mock this `$source = 'default'` by using `Magento\InventoryCatalogApi\Api\DefaultSourceProviderInterface`? Please take a look I've updated that in my latest commit.